### DISCUSSION TITLE
posts/terraform-cloudflare-github/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#249] [Comments]
# posts/terraform-cloudflare-github/

Today, we’re going to set up and configure Terraform on your machine so we can start using Terraform. Then we’ll configure cf-terraforming to import our Cloudflare state and configuration into Terraform. After that we’ll set up a GitHub report and configure GitHub actions so you have CI and CD for deploying your Infrastructure automatically using a Git Flow. If you’re new to Terraform, that’s fine! This is a beginner tutorial for Terraform and by the end of this, you will feel like an expert!

https://technotim.live/posts/terraform-cloudflare-github/

**Comment by sachasmart** [comment]
Awesome video, Tim!

Just one addition. Like many homelabbers, I use  oznu/cloudflare-ddns to dynamically update my cloudflare A record's IP address. 

With the current config, the terraform state is static and could potentially update to an old IP address if the homelab's IP address changes in the interim. To ensure that the ip address of cloudflare always matches the current IP address of the homelab within the terraform state you can add the following block to `data.tf`:

```
data "http" "homelab_ip" {
  url = "https://api.ipify.org"
}
```
then call this resource in your cloudflare A record resource:
```
resource "cloudflare_record" "terraform_managed_resource_<uuid>" {
  name    = "<name>"
  proxied = true
  ttl     = 1
  type    = "A"
  value   = data.http.homelab_ip.response_body <--- add this block
  zone_id = "<zone_id>"
}
```

Hope someone finds this useful. Cheers!

**Comment by oleksdovz** [comment]
Why U don't use tunnel from Cloudflare? For me it's next stage for homeLab

**Reply by timothystewart6** [reply]
I like a single point of entry through my reverse proxy where I can inspect all traffic.
